packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -39,6 +41,20 @@ export interface IDistribution extends IResource {
    * @attribute
    */
   readonly distributionId: string;
+
+  /**
+   * The distribution ARN for this distribution.
+   *
+   * @attribute
+   */
+  readonly distributionArn: string;
```

Comment:
I'm not sure I get why this is being added in this context. It's not included in `Fn::GetAtt` in the resource per https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-distribution.html. How is this being used?

Reply:
It's useful when grant other than `CreateInvalidation`:

```ts
declare const role: iam.Role;

// before
role.addToPrincipalPolicy(new iam.PolicyStatement({
  actions: ['cloudfront:ListInvalidations'],
  // region is needed here
  resources: [this.formatArn({ service: 'cloudfront', region: '', resource: 'distribution', resourceName: dist.distributionId })],
}));

// after
role.addToPrincipalPolicy(new iam.PolicyStatement({
  actions: ['cloudfront:ListInvalidations'],
  resources: [dist.distributionArn],
}));
```

Can be removed if it's excessive.